src/string/isomorphism.rs

```diff
@@ -0,0 +1,59 @@
+/*
+Given two strings s and t, determine whether they are isomorphic.
+The two strings are isomorphic if the characters in s can be replaced by some mapping relation to get t
+*/
+use std::collections::HashMap;
+
+pub fn is_isomorphic(s: &str, t: &str) -> bool {
+    let s_chars: Vec<char> = s.chars().collect();
+    let t_chars: Vec<char> = t.chars().collect();
+    if s_chars.len() != t_chars.len() {
+        return false;
+    }
+    let mut s_to_t_map = HashMap::new();
+    let mut t_to_s_map = HashMap::new();
+    for (s_char, t_char) in s_chars.into_iter().zip(t_chars) {
+        if !check_mapping(&mut s_to_t_map, s_char, t_char)
+            || !check_mapping(&mut t_to_s_map, t_char, s_char)
+        {
+            return false;
+        }
+    }
+    true
+}
+fn check_mapping(map: &mut HashMap<char, char>, key: char, value: char) -> bool {
```

Comment:
```suggestion

/// Checks the mapping between two characters and updates the map.
///
/// # Arguments
///
/// * `map` - The HashMap to store the mapping.
/// * `key` - The key character.
/// * `value` - The value character.
///
/// # Returns
///
/// `true` if the mapping is consistent, `false` otherwise.
fn check_mapping(map: &mut HashMap<char, char>, key: char, value: char) -> bool {
```